UnityPy/files/ObjectReader.py

```diff
@@ -200,13 +200,13 @@ def dump_typetree_structure(self) -> str:
 
     def read_typetree(self, nodes: list = None) -> dict:
         self.reset()
+        tree = dict()
         if nodes:
             tree = TypeTreeHelper.read_typetree(nodes, self)
         elif getattr(self.serialized_type, "nodes", None):
-            tree = TypeTreeHelper.read_typetree(
-                self.serialized_type.nodes, self)
-        else:
-            tree = {}
+            if self.serialized_type.nodes:
```

Comment:
I backtracked a bit and noticed that serialized_type should always have the field nodes,
so instead of
```python
 elif getattr(self.serialized_type, "nodes", None):
     if self.serialized_type.nodes:
```

```python
elif self.serialized_type.nodes:
```
should do.